src/cider/nrepl/middleware/ns.clj

```diff
@@ -51,6 +58,43 @@
   (transport/send transport (response-for msg :ns-vars (ns-vars msg)))
   (transport/send transport (response-for msg :status :done)))
 
+(defn- jar-file?
+  "Returns true if file is a normal file with a .jar or .JAR extension."
```

Comment:
I stole this function from `tools.namespace`, where it's private, so I assume so :(